### DISCUSSION TITLE
EWMH compatible state listener

### DISCUSSION THREAD

**Discussion by Philipp-M** [elkowar/eww#705] [Show and tell]
Hey, since it took a little bit more effort than initially anticipated (probably because I *had* to do it in Rust directly with an XCB connection instead of Rust/xprop or Bash/xprop, lol, but it's fast now at least (<1ms for a change)) and it may be helpful for others as well:

I wrote a simple EWMH status listener, which monitors the current state of an EWMH compatible window manager and outputs it as one-line json each time the EWMH state changes (e.g. the current workspace is changed or windows get opened/closed/focused etc.).

With this the need for a separate status bar is gone IMHO (which was also the motivation for this):

https://github.com/Philipp-M/ewmh-status-listener

**Comment by musjj** [comment]
Nice, I can throw out my hacky bash script now. IMHO, this kind of stuff should be integrated into eww. We already have stuff like network, ram, etc. built-in anyways.

**Reply by Philipp-M** [reply]
Yeah you're right. I just have seen, that there are these builtin "Magic Variables". It won't be as trivial as a new Poll source though, since there aren't any `deflisten` builtin "Magic Variables" yet. But I guess it shouldn't be too hard anyway to add this as a core feature, I guess I can create a PR.

Btw. I have just updated the listener, as I had crashes very rarely (I have not really identified the source yet, but I guess sometimes the xcb answers are weird).
 if anything goes wrong, everything will be restarted in a throttled loop.

**Reply by Philipp-M** [reply]
I've created an issue for that: https://github.com/elkowar/eww/issues/735
